dask/dataframe/optimize.py

```diff
@@ -20,8 +13,10 @@ def optimize(dsk, keys, **kwargs):
     try:
         from castra import Castra
         dsk3 = fuse_getitem(dsk2, Castra.load_partition, 3)
+        dsk4 = fuse_selections(dsk3, getattr, Castra.load_partition,
+                               lambda a, b: (Castra.load_index, b[1], b[2]))
```

Comment:
This seems a bit broad of a condition.  Perhaps `.index` should use a special function like 

```python
def get_index(df):
    return f.index
```

And then we can use that in place of `getattr`